fix(isolated-declarations): leading comments of `ExportDefaultDeclaration` and `TSExportAssignment` appear in incorrect places

Dunqing opened the pull request:
close: #10535


https://www.typescriptlang.org/play/?#code/PQKhCgAIUgTBTAZgQwK4BsAul4A8AOA9gE6ZQjDh5GlxJpaQCMADOOKBNJACoDKAUQIlMAQQDO4gJYBzAHYBbeHLLRKVYbQC8kRKjkBjTFMJzdhQgAoAlJADekYvEypiZlgF8gA

Boshen commented:
### Merge activity

* **Apr 23, 4:51 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(isolated-declarations): leading comments of `ExportDefaultDeclaration` and `TSExportAssignment` appear in incorrect places (#10559)

close: #10535

https://www.typescriptlang.org/play/?#code/PQKhCgAIUgTBTAZgQwK4BsAul4A8AOA9gE6ZQjDh5GlxJpaQCMADOOKBNJACoDKAUQIlMAQQDO4gJYBzAHYBbeHLLRKVYbQC8kRKjkBjTFMJzdhQgAoAlJADekYvEypiZlgF8gA
